### ISSUE TITLE
JSON value '<null>' of type NSNull cannot be converted to CLLocationDegrees

### ISSUE THREAD

**Issue by burakkbilginn** [facebook/react-native#33489]
### Description

My react-native app is working fine in android without any problem. But when I run it in IOS simulator, this error comes to app screen and it doesn't go further:
```
JSON value '<null>' of type NSNull cannot be converted to CLLocationDegrees 

+ RCTConvert(CoreLocation) CLLocationDegrees:] RCTConvert+CoreLocation.m:12 

+ RCTConvert(CoreLocation) CLLocationCoordinate2D:] 

+ [RCTConvert(AirMap) MKCoordinateRegion:] 

__invoking___ 

-[NSInvocation invoke] 

__41-[RCTModuleMethod processMethodSignature]_block_invoke.80 

__41-[RCTModuleMethod processMethodSignature]_block_invoke.164 -[RCTModuleMethod invokeWithBridge:module:arguments:] 

facebook::react::invokeInner(RCTBridge*, RCTModuleData*, unsigned int, folly::dynamic const&, int, (anonymous namespace)::SchedulingContext) 

facebook::react::RCTNativeModule::invoke(unsigned int, folly::dynamic&&, int)::$_0::operator()() 

const invocation function for block in facebook::react::RCTNativeModule::invoke(unsigned int, folly::dynamic&&, int)

_dispatch_call_block_and_release 

_dispatch_client_callout 

_dispatch_lane_serial_drain
    
_dispatch_lane_invoke 

_dispatch_workloop_worker_thread 

_pthread_wqthread 

start_wqthread
```

Here are the node modules:

```
"dependencies": {
    "@react-native-async-storage/async-storage": "^1.17.0",
    "@react-native-community/cli": "^7.0.3",
    "@react-native-community/geolocation": "^2.0.2",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "^8.2.0",
    "@react-native-picker/picker": "^2.4.0",
    "@react-navigation/drawer": "^6.3.1",
    "@react-navigation/native": "^6.0.8",
    "@react-navigation/stack": "^6.1.1",
    "amazon-cognito-identity-js": "^5.2.8",
    "aws-amplify": "^4.3.12",
    "aws-amplify-react-native": "^6.0.2",
    "jetifier": "^2.0.0",
    "react": "17.0.2",
    "react-native": "^0.67.4",
    "react-native-gesture-handler": "2.1.1",
    "react-native-google-places-autocomplete": "^2.4.1",
    "react-native-maps": "0.30.1",
    "react-native-maps-directions": "^1.8.0",
    "react-native-permissions": "^3.3.1",
    "react-native-reanimated": "2.4.1",
    "react-native-safe-area-context": "^4.2.1",
    "react-native-screens": "^3.13.1",
    "react-native-vector-icons": "^9.1.0",
    "react-navigation": "^4.4.4",
    "react-promise": "^3.0.2",
    "uuid": "^8.3.2"
  },
```

### Version

0.67.4

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-3317U CPU @ 1.70GHz
    Memory: 611.86 MB / 12.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 17.6.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.5.1 - /usr/local/bin/npm
    Watchman: 2022.02.28.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: ^7.0.3 => 7.0.3 
    react: 17.0.2 => 17.0.2 
    react-native: ^0.67.4 => 0.67.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Since the app is working fine in android, I don't think the issue is about the coding part. Maybe it is about the environment part.

**Comment by mensonones** [comment]
Hey, @burakkbilginn!  This error is probably from the library react-native-maps or @react-native-community/geolocation

**Comment by burakkbilginn** [comment]
> Hey, @burakkbilginn! This error is probably from the library react-native-maps or @react-native-community/geolocation

So is there a way to handle this? Maybe downgrade to a specific version or any other method?

**Comment by burakkbilginn** [comment]
After some time, I noticed that onMapReady event doesn't get fired in IOS simulator, therefore the coordinates aren't grasped and stays null. This issue is being told to start appearing after last google maps update, seem to be about 10 days ago, here mentioned: 
[](https://github.com/react-native-maps/react-native-maps/issues/4066#issue-1102036177)

I digged in google but couldn't find a working workaround. Any help is appreciated.
